### PR TITLE
[native] Add sidecar system property

### LINKED ISSUE #22829: Add Presto sidecar process type
To support [RFC-0003](https://github.com/prestodb/rfcs/blob/main/RFC-0003-native-spi.md), a new process type is introduced, the sidecar.  The purpose of the sidecar is to provide a series of endpoints to inform the Presto coordinator of the capabilities of the underlying workers, and to provide a mechanism for the coordinator to use that functionality.  In order to support constant folding without relying on coincidental implementations of user-defined functions in Java, for example, either expressions or functions must be executed in C++, and the sidecar can support this.  Likewise, the sidecar can be used to inform the coordinator of the function registry and worker-level session properties.

This issue is to add support for the Presto sidecar.  It will be implemented as a new process type in Presto, alongside the worker, coordinator, and resource manager.  The sidecar will announce itself to the discovery service, as other services do.  The coordinator will have some small changes made to change its behavior when the sidecar is expected but not running, for example, it may queue or fail queries.

The sidecar will be externally managed, as with other process types in Presto.  It is expected that external orchestration will ensure the sidecar is running.  Eventually, the Presto Helm chart will be updated for native clusters to utilize the sidecar, and external users must eventually opt to use the sidecar.

The code for the sidecar will be in the existing `presto-native-execution` module and will share code with the C++ worker process developed there.

### PR DESCRIPTION
**Pull request by pramodsatya**
## Description

Adds system property for native sidecar.
Adds process type `sidecar` to announcement.

Resolves #22829 

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by amitkdutta — COMMENTED** [review]
Register the property here https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/common/Configs.cpp#L182
Its not centralized which needs improvement.

Thanks @pramodsatya. Looks good. Few nits.

**Comment by pramodsatya** [comment]
> Register the property here https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/common/Configs.cpp#L182 Its not centralized which needs improvement.
> 
> Thanks @pramodsatya. Looks good. Few nits.

Thanks for the feedback @amitkdutta, moved the property registration here. Could you please elaborate on what is not centralized?

**Comment by amitkdutta** [comment]
> > Register the property here https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/common/Configs.cpp#L182 Its not centralized which needs improvement.
> > Thanks @pramodsatya. Looks good. Few nits.
> 
> Thanks for the feedback @amitkdutta, moved the property registration here. Could you please elaborate on what is not centralized?

Ideally introducing a config property should automatically register it, extra registration (like the one you did here) should not be needed. But current framework does not support it, thus defining config and registering is not centralized. Nothing related to this PR though, was just mentioning a weakness of current framework.

**Review by amitkdutta — APPROVED** [review]
Looks good. Thanks @pramodsatya for adding the change.

File: presto-native-execution/presto_cpp/main/common/Configs.h [hunk 1]
```diff
@@ -248,6 +248,7 @@ class SystemConfig : public ConfigBase {
   /// Memory allocation limit enforced via internal memory allocator.
   static constexpr std::string_view kSystemMemoryGb{"system-memory-gb"};
 
+  static constexpr std::string_view kNativeSidecar{"native-sidecar"};
```
Comment (aditi-pandit) [inline presto-native-execution/presto_cpp/main/common/Configs.h:251]: Please can you add a comment explaining this config.

**Review by aditi-pandit — COMMENTED** [review]
Thanks @pramodsatya. Minor comment.

File: presto-native-execution/presto_cpp/main/common/Configs.h (hunk 1 above)
Comment (pramodsatya) [reply presto-native-execution/presto_cpp/main/common/Configs.h:252]: Added a comment, thanks.

Comment (aditi-pandit) [reply presto-native-execution/presto_cpp/main/common/Configs.h:252]: @pramodsatya : Curious: Is this meant to be "can function" as a sidecar, or "is a" sidecar ?

Comment (pramodsatya) [reply presto-native-execution/presto_cpp/main/common/Configs.h:252]: @aditi-pandit the plan is to register only the sidecar specific endpoints when this flag is enabled on the worker. Can I reword it as "Indicates if the worker is configured as a sidecar." ?

Comment (aditi-pandit) [reply presto-native-execution/presto_cpp/main/common/Configs.h:252]: @pramodsatya : That sounds reasonable.

Comment (tdcmeehan) [reply presto-native-execution/presto_cpp/main/common/Configs.h:252]: "Indicates if the process is configured as a sidecar."

Comment (pramodsatya) [reply presto-native-execution/presto_cpp/main/common/Configs.h:252]: Updated, thanks.

**Review by aditi-pandit — APPROVED** [review]
Thanks @pramodsatya

**Pull request merged**